ir/ir.def

```diff
@@ -178,6 +178,25 @@ class KeyElement {
     Annotations     annotations;
     Expression      expression;
     PathExpression  matchType;
+
+    const IR::Node *transform_visit(Transform &v) {
```

Comment:
I expect most backends to need to do this in at least one pass, if not multiple.  Ideally this would have been handled automatically by the ir-generator, but I'm not sure how to do that -- so this overrides the default autogenerated visit_children, but must be called explicitly from the visitor preorder function to work.